### DISCUSSION TITLE
Need a way to analyse what is causing HBase table size to grow very quickly

### DISCUSSION THREAD

**Discussion by jayendrap** [JanusGraph/janusgraph#3304] [Q&A]
I use Hbase as the storage backend for janusgraph, but the size of HBase table is growing very quickly (within 4 months size increased from 16 GB to 280GB ), even when there isn't much vertices and edges getting created. 
I am trying to run some gremlin queries to find if there are vertices having attributes which has large values, but there weren't any large values as well.
I tried querying HBase table, but the information wasn't readable as plain text.
Is there a way to determine what is causing table size to grow ?

**Comment by vtslab** [comment]
This is mysterious, indeed. First thing that comes to mind: is it possible that compaction is disabled on your HBase cluster (hbase.regionserver.compaction.enabled)? What happens if you [trigger a manual compaction](https://community.cloudera.com/t5/Support-Questions/how-to-run-major-compaction-and-how-to-monitor-them-while/td-p/214737)?

**Reply by jayendrap** [reply]
@vtslab Thanks for your response. I will check that property.
Also, if I delete some vertices and edges and then run the compaction manually, then it frees up some space, but that is not significant.  So I wanted to check which are the vertices and edges which are consuming most of the space, so that I can try deleting them to free up some space, or apart from vertices and edges, is there any thing else that is taking up the space?

**Reply by porunov** [reply]
Not sure, but also could be a transaction log enabled without clearing up old logs. So, any mutation would result in more and more data. That said, by default it's disabled, so I'm not sure if it's the problem.

**Reply by jayendrap** [reply]
Thanks @porunov , I checked the configuration, it seems tx-logs are enabled in my case. I am checking the hbase data to see if indeed tx-logs is consuming that much data.
